tests/subsys/fs/nvs/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  filesystem.nvs:
+    platform_whitelist: nrf51_pca10028
```

Comment:
I have a feeling we should are more platforms here. What are the flash access requirements of NVS? Does it require unaligned accesses like NFFS?

Reply:
@carlescufi , this is the only platform I have, and the only platform I have tested it on. There are no special flash requirements from NVS, it should work on any flash. NVS dies not require unaligned access, all flash writes are sequentially (no rewrites to allready written flash locations).

Reply:
@MaureenHelm Can you run this test on frdm_k64f, just want to know whether NVS works on the device with write-block-size of 8 bytes.

Reply:
@MaureenHelm, if you run the test don't forget to change the sector size to match your platform. As the test is written it uses a page size of 1kB.

Reply:
@Laczen thanks for the input. I will run this on nRF52x and then you can add those to the list

Reply:
As said in the other bug report, it partially fixes things from me, but I don't think it's the right way to fix that. The __aligned(FLASH_WRITE_BLOCK_SIZE) directive only means that the beginning of the structure has to be aligned **in RAM** to the write size. It doesn't change the size of the structure, so it means write methods are still writing things out of stack to the flash and that the stack is corrupted on reads. Also it doesn't fix writing values smaller to the block size.

In addition I am not sure using FLASH_WRITE_BLOCK_SIZE is correct here. From my understanding, it corresponding to the block size of the SoC. It doesn't necessary match the write size of an SPI flash, and we might want to use NVS on it (I don't know if it is something possible yet).

One other way to go is to change nvs_flash_write() to pad the data to write with 0, and to not use _nvs_len_in_flash for reads. I'll try to work on a patch doing that.